repositories/migrations.go

```diff
@@ -17,63 +19,101 @@ import (
-		return nil, fmt.Errorf("unable to connect to database: %w", err)
+func (m *Migrater) Run(ctx context.Context) error {
+	if err := m.openDb(); err != nil {
+		return errors.Wrap(err, "unable to open db in Migrater")
 	}
 
-	err = migrationDB.Ping()
-	if err != nil {
-		return nil, fmt.Errorf("unable to ping database: %w", err)
+	// Now run the migrations
+	if err := m.runMarbleDbMigrations(ctx); err != nil {
+		return errors.Wrap(err, "runMarbleDbMigrations error")
 	}
 
-	return migrationDB, nil
+	if err := m.migrateAnalyticsViews(ctx, m.analyticsViewsFileSystem); err != nil {
+		return errors.Wrap(err, "migrateAnalyticsViews error")
+	}
+
+	return nil
 }
 
-func RunMigrations(env string, pgConfig utils.PGConfig, logger *slog.Logger) error {
-	db, err := setupDbConnection(env, pgConfig)
+func (m *Migrater) openDb() error {
+	connectionString := m.pgConfig.GetConnectionString(m.env)
+	db, err := sql.Open("pgx", connectionString)
 	if err != nil {
-		return fmt.Errorf("setupDbConnection error: %w", err)
-	}
-
-	params := migrationParams{
-		fileSystem:   embedMigrations,
-		folderName:   "migrations",
-		allowMissing: false,
+		return errors.Wrap(err, "unable to create connection pool for migrations")
+	} else {
+		m.db = db
 	}
-	if err := runMigrationsWithFolder(db, params, logger); err != nil {
-		return fmt.Errorf("runMigrationsWithFolder error: %w", err)
+	if err = m.db.Ping(); err != nil {
+		return errors.Wrap(err, "unable to ping database")
 	}
 	return nil
 }
 
-func runMigrationsWithFolder(db *sql.DB, params migrationParams, logger *slog.Logger) error {
-	// start goose migrations
-	logger.Info("Migrations starting to setup DB: " + params.folderName)
-	goose.SetBaseFS(params.fileSystem)
-
+func (m *Migrater) runMarbleDbMigrations(ctx context.Context) error {
+	logger := utils.LoggerFromContext(ctx)
+	logger.InfoContext(ctx, "Migrations starting to setup marble DB")
+	goose.SetBaseFS(m.dbMigrationsFileSystem)
 	if err := goose.SetDialect("postgres"); err != nil {
 		return err
 	}
+	return goose.Up(m.db, "migrations")
+}
 
-	// When running on the secondary folder containing the test org migrations, we allow missing migrations to allow out of order migrations with the main folder
-	if params.allowMissing {
-		if err := goose.Up(db, params.folderName, goose.WithAllowMissing()); err != nil {
-			return fmt.Errorf("unable to run migrations: %w", err)
-		}
-	} else {
-		if err := goose.Up(db, params.folderName); err != nil {
-			return fmt.Errorf("unable to run migrations: %w", err)
-		}
+func (m *Migrater) migrateAnalyticsViews(ctx context.Context, folder embed.FS) error {
+	if err := fs.WalkDir(
+		folder,
+		".",
+		func(path string, d fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if !d.IsDir() {
+				return m.createViewFromFile(ctx, folder, path)
+			}
+
+			return nil
+		},
+	); err != nil {
+		return errors.Wrap(err, "error while walking embedded analytics_views folder")
+	}
+	return nil
+}
```

Comment:
What about cleaning ? IIUC, we need to let at least a file with `DROP VIEW IF EXISTS ...` to ensure suppression of deprecated views is that right ?

Reply:
Hmm, you mean if we remove a table or such, right 🤔 

Actually, most likely we need to drop the view before we drop the corresponding table, so we'd be kinda forced to do this in the migration file anyway